packages/widgets/user-profile-widget/src/lib/widget/mixins/initMixin/initComponentsMixins/initUserCustomAttributesMixin.ts

```diff
@@ -0,0 +1,46 @@
+import { UserAttributeDriver } from '@descope/sdk-component-drivers';
+import { compose, createSingletonMixin } from '@descope/sdk-helpers';
+import { loggerMixin } from '@descope/sdk-mixins';
+import { getUserCustomAttrs } from '../../../state/selectors';
+import { stateManagementMixin } from '../../stateManagementMixin';
+import { initWidgetRootMixin } from './initWidgetRootMixin';
+
+export const initUserCustomAttributesMixin = createSingletonMixin(
+  <T extends CustomElementConstructor>(superclass: T) =>
+    class UserCustomAttributesMixinClass extends compose(
+      stateManagementMixin,
+      loggerMixin,
+      initWidgetRootMixin,
+    )(superclass) {
+      customValueUserAttr: UserAttributeDriver;
+
+      #initCustomValueUserAttrs() {
```

Comment:
Now there is no need to call the selector inside the fn
you get the data in the fn params
In addition, we should wrap it with `withMemCache` 
See example [here](https://github.com/descope/descope-js/blob/f159ae8ebd6ac37c9e707252f5bf16e73b144171/packages/widgets/user-profile-widget/src/lib/widget/mixins/initMixin/initComponentsMixins/initAvatarMixin.ts#L74)

If something is not clear we can do it together

Reply:
@nirgur , tested it just now and seems like I must have the first call (otherwise the values are not being presented).

Added the caching, please check.